src/pkg/services/m365/api/graph/errors.go

```diff
@@ -133,16 +133,38 @@ func stackWithCoreErr(ctx context.Context, err error, traceDepth int) error {
 		err = clues.Stack(core.ErrAlreadyExists, err)
 	case isErrNotFound(ode, err):
 		err = clues.Stack(core.ErrNotFound, err)
+	default:
+		err = toErrByRespCode(ode, err)
 	}
 
 	return stackWithDepth(ctx, err, 1+traceDepth)
 }
 
 // unexported categorizers, for use with stackWithCoreErr
 
+// for categorizing errors by their response code and no other response propery
+func toErrByRespCode(ode oDataErr, err error) error {
+	fiveXX := []int{}
+	for i := 0; i < 5; i++ {
+		fiveXX = append(fiveXX, 500+i)
```

Comment:
nit: Having `fiveXX` as a const would be better here. Avoids new allocations.

Reply:
Per recommendation, I ditched the slice and am now catching all 5xx errors.